### PR TITLE
Use npm-lockdown to install Node dependencies

### PR DESCRIPTION
**Pull request by rehandalal**
r?

### CODE REVIEW

File: scripts/relock.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env node
```
Comment (mythmon) [inline scripts/relock.js]: Can we name this file `lockdown-relock.js`? Or would that make things weird?

Comment (rehandalal) [inline scripts/relock.js]: that should be fine. i was also waffling on including this file. i thought maybe it made more sense to add lockdown as a dev dependency and install it and the relock bin from there.

Comment (mythmon) [inline scripts/relock.js]: If we have lockdown's hash in the thing, and it installs itself, so we can trust it, i think that would work. We definitely need at least lockdown.js in the repo.

Comment (mythmon) [inline scripts/relock.js]: And by "in the thing" I mean "in lockdown.json"

**Comment by mythmon** [comment]
I like it. I like that it is so much simpler than peep was. Erirk Rose loves it. We should test it on stage before we merge this. Lets decide about the relock.js file, test this on stage, and then land it. :fireworks:

**Comment by rehandalal** [comment]
updated ^

**Comment by rehandalal** [comment]
docs updated ^

**Comment by mythmon** [comment]
Wow, we have a lot of node dependencies from all our dependencies' dependencies.

I tested what happens if a hash is bad or missing. In both cases I got this:

```$ npm install
> Kitsune@ preinstall /home/mythmon/src/kitsune
> ./scripts/lockdown.js

NPM Lockdown is here to check your dependencies!  Never fear!
using registry: registry.npmjs.org

> Kitsune@ preinstall /home/mythmon/src/kitsune
> ./scripts/lockdown.js

npm WARN optional dep failed, continuing fsevents@0.2.1
npm ERR! Linux 3.18.6-1-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! node v0.10.36
npm ERR! npm  v2.5.0
npm ERR! code ETARGET

npm ERR! notarget No compatible version found: abbrev@'>=1.0.0 <2.0.0'
npm ERR! notarget No valid targets found.
npm ERR! notarget Perhaps not compatible with your version of node?
npm ERR! notarget This is most likely not a problem with npm itself.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/mythmon/src/kitsune/npm-debug.log

npm ERR! Linux 3.18.6-1-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! node v0.10.36
npm ERR! npm  v2.5.0
npm ERR! code ELIFECYCLE
npm ERR! Kitsune@ preinstall: `./scripts/lockdown.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the Kitsune@ preinstall script './scripts/lockdown.js'.
npm ERR! This is most likely a problem with the Kitsune package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ./scripts/lockdown.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls Kitsune
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/mythmon/src/kitsune/npm-debug.log
```

This also returned an exit code of 1, which means it will fail in Travis and in deploy, which is good. It is a bit confusing though, so we should add it to the docs. We should also add a note about how to add new dependencies to the docs.

Besides the doc changes, I think everything looks good.

**Comment by mythmon** [comment]
Doc changes look good. r+

**Pull request merged**